homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,688 @@
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensordata: dict,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            sensordata=sensordata,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the pump speed sensor."""
+        _LOGGER.debug("Updating state of sensor: %s", self._name)
+
+        sensordata = {}
+        pump_type = ""
+        pump_speed = 0
+
+        if self._kind == "filter_pump_speed":
+            for backyard in self.coordinator.data:
+                for bow in backyard.get("BOWS"):
+                    if bow["Filter"]["systemId"] == self.entitydata.get("systemId"):
+                        sensordata = bow["Filter"]
+                        break
+
+            if sensordata.get("Filter-Type") == "FMT_VARIABLE_SPEED_PUMP":
+                pump_type = "VARIABLE"
+            elif sensordata.get("Filter-Type") == "FMP_SINGLE_SPEED":
+                pump_type = "SINGLE"
+            elif sensordata.get("Filter-Type") == "FMT_DUAL_SPEED":
+                pump_type = "DUAL"
+
+            pump_speed = sensordata.get("filterSpeed")
+
+        elif self._kind == "pump_speed":
+            for backyard in self.coordinator.data:
+                for bow in backyard.get("BOWS"):
+                    for pump in bow.get("Pumps"):
+                        if pump["systemId"] == self.entitydata["systemId"]:
+                            sensordata = pump
+                            break
+
+            if sensordata.get("Filter-Type") == "PMP_VARIABLE_SPEED_PUMP":
+                pump_type = "VARIABLE"
+            elif sensordata.get("Filter-Type") == "PMP_SINGLE_SPEED":
+                pump_type = "SINGLE"
+            elif sensordata.get("Filter-Type") == "PMP_DUAL_SPEED":
+                pump_type = "DUAL"
+
+            pump_speed = sensordata.get("pumpSpeed")
+
+        if pump_type == "VARIABLE":
+            self._unit = "%"
+            self._state = pump_speed
+        elif pump_type == "DUAL":
+            if pump_speed == 0:
+                self._state = "off"
+            elif pump_speed == sensordata.get("Min-Pump-Speed"):
+                self._state = "low"
+            elif pump_speed == sensordata.get("Max-Pump-Speed"):
+                self._state = "high"
+        elif pump_type == "SINGLE":
+            if pump_speed == 0:
+                self.state = "off"
+            elif pump_speed == sensordata.get("Max-Pump-Speed"):
+                self._state = "on"
+
+        self._name = (
+            self._backyard["BackyardName"]
+            + " "
+            + self.bow.get("Name")
+            + " "
+            + self.entitydata.get("Name")
```

Comment:
Store the name from the entity data as an instance attribute already in the parent entity class.

Reply:
Similar comment here - can you clarify the request?

Reply:
OK - addressed properly throughout.